include/boost/thread/future.hpp

```diff
@@ -1133,7 +1126,8 @@ namespace boost
                 {
                     for(count_type_portable i=0;i<count;++i)
                     {
-                        locks[i].unlock();
+                        if (locks[i].owns_lock())
```

Comment:
Hi, have you found a case where the lock where not locked?

Reply:
@viboes I've completely missed this function is expected to `unlock`